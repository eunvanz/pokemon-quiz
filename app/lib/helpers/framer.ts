import { Variants } from 'framer-motion'

export const motionVariants: Variants = {
  vibe: {
    translateX: [0, -4, 0, 4, 0],
    transition: {
      repeat: 2,
      duration: 0.1,
    },
  },
}
