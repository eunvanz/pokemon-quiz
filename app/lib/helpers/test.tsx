import CommonProvider from '@/app/components/common-provider'
import { PropsWithChildren } from 'react'
import api from '../api'

export interface TestProviderProps {
  api?: Partial<typeof api>
}

export const TestProvider: React.FC<PropsWithChildren<TestProviderProps>> = ({
  children,
  api = {},
}) => <CommonProvider api={api}>{children}</CommonProvider>
