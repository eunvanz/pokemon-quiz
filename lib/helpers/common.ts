export const checkIsSSR = () => {
  return typeof window === 'undefined'
}
