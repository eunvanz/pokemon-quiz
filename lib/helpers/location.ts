import { countries, zones } from 'moment-timezone/data/meta/latest.json'

export const getTimeZoneToCountryMap = () => {
  const timeZoneToCountryMap: Record<string, { name: string; abbr: string }> =
    {}
  Object.keys(zones).forEach((zone) => {
    const cityArr = zone.split('/')
    const city = cityArr[cityArr.length - 1]
    // @ts-ignore
    timeZoneToCountryMap[city] = countries[zones[zone].countries[0]]
  })
  return timeZoneToCountryMap
}

export const getUserLocationFromTimeZone = () => {
  const timeZone = Intl?.DateTimeFormat().resolvedOptions().timeZone
  if (timeZone) {
    const tzArr = timeZone.split('/')
    const region = tzArr[0]
    const city = tzArr[tzArr.length - 1]
    const { name, abbr } = getTimeZoneToCountryMap()[city]
    return {
      city,
      country: name,
      countryCode: abbr,
      region,
    }
  } else {
    return undefined
  }
}

export const getCountryFlagImgUrl = (countryCode: string) => {
  return `http://purecatamphetamine.github.io/country-flag-icons/3x2/${countryCode.toUpperCase()}.svg`
}
