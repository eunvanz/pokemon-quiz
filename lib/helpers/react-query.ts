import { QueryClient } from 'react-query'
import { Pageable } from '../types'

export const queryClient = new QueryClient()

export const getMergedPageData = <T>(data: Pageable<T>[]) => {
  return data.reduce((prev: T[], item) => [...prev, ...item.items], [])
}
