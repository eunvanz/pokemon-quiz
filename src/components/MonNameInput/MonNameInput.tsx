import { useCallback } from "react";
import { useForm } from "react-hook-form";
import tw from "twin.macro";
import Button from "../Button";
import TextField from "../TextField";

export interface MonNameInputProps {
  onSubmit: (name: string) => void;
  correctAnswer: string;
}

const MonNameInput: React.FC<MonNameInputProps> = ({ onSubmit, correctAnswer }) => {
  const { register, handleSubmit, setError } = useForm();

  const handleOnSubmit = useCallback(({ monName }) => {}, []);

  return (
    <form onSubmit={handleSubmit(handleOnSubmit)}>
      <div css={tw`flex`}>
        <div css={tw`flex-1 pr-2`}>
          <TextField isBlock placeholder="Enter PokÃ©mon's name" />
        </div>
        <Button css={tw`px-6`}>Fire (Enter)</Button>
      </div>
      <div css={tw`pt-2`}>
        <Button css={tw`w-full`} color="secondary">
          Skip (Space bar)
        </Button>
      </div>
    </form>
  );
};

export default MonNameInput;
