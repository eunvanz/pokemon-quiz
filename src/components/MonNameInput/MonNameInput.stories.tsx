import type { ComponentStory, ComponentMeta } from "@storybook/react";
import { createStoryComponent } from "~/helpers/storybookHelpers";
import MonNameInput from "./MonNameInput";

export default {
  title: "components/MonNameInput",
  component: MonNameInput,
  args: {
    correctAnswer: "ditto",
  },
} as ComponentMeta<typeof MonNameInput>;

const Template: ComponentStory<typeof MonNameInput> = (args) => (
  <MonNameInput {...args} />
);

export const Default = createStoryComponent(Template);
