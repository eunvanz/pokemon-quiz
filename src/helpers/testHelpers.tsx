import { Story } from "@storybook/react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClientProvider } from "react-query";
import api from "~/api";
import { ApiProvider } from "~/api/apiContext";
import { queryClient } from "./reactQuery";

/**
 * render storybook component in test code
 * @param StoryComponent
 * @param props
 * @returns
 */
export const renderStory = <T,>(
  StoryComponent: Story<Partial<T>>,
  props?: Partial<T>,
  renderOptions?: RenderOptions,
) => {
  return render(<StoryComponent {...props} />, renderOptions);
};

export interface TestProviderProps {
  api: Partial<typeof api>;
}

export const TestProvider: React.FC<TestProviderProps> = ({ children, api }) => (
  <ApiProvider api={api}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ApiProvider>
);
