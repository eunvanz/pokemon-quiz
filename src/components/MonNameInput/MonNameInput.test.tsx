import { composeStories } from "@storybook/testing-react";
import { screen } from "@testing-library/react";
import { renderStory } from "~/helpers/testHelpers";
import { MonNameInputProps } from "./MonNameInput";
import * as stories from "./MonNameInput.stories";

const { Default } = composeStories(stories);

describe("MonNameInput", () => {
  describe("Default", () => {
    const setup = (props?: Partial<MonNameInputProps>) => renderStory(Default, props);

    it("matches with previous snapshot", () => {
      const { container } = setup();
      expect(container).toMatchSnapshot();
    });

    describe("when input is empty", () => {
      it("can not be submitted", () => {});
    });

    describe("when wrong answer is submitted", () => {
      it("resets input and shows error message", () => {});
    });

    describe("when correct answer is submitted", () => {
      it("calls [onSubmit]", () => {});
    });
  });
});
