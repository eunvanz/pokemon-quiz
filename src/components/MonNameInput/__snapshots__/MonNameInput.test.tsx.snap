// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonNameInput Default matches with previous snapshot 1`] = `
<div>
  <form>
    <div
      css="[object Object]"
    >
      <div
        css="[object Object]"
      >
        <div
          css="[object Object]"
        >
          <input
            css="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],"
            placeholder="Enter PokÃ©mon's name"
          />
          <caption
            css="[object Object],[object Object]"
          />
        </div>
      </div>
      <button
        css="[object Object]"
      >
        Fire (Enter)
      </button>
    </div>
    <div
      css="[object Object]"
    >
      <button
        css="[object Object]"
      >
        Skip (Space bar)
      </button>
    </div>
  </form>
</div>
`;
